docs/network-layer.md

```diff
@@ -0,0 +1,61 @@
+
+## Control Flow
+When the node is running, it performs several housekeeping tasks wrt. networking related matters. More specifically the node checks that its current connections are *healthy* i.e., verifying reasonable latency, removing connections of which a handshake is missing etc. If any connections deem *unhealthy* then the node will close the connection and remove the peer from its set of peers. Note that depending on the level of *unhealthy* the node will perhaps *soft ban* a peer (more on this below).
+
+If the node has exceeded its maximum amount of peers it will remove a random subset of its peers. Note that the node will never remove any of the *pinned peers*. 
+The node then proceeds to check if it should unban any of the *soft banned* peers and does so if that is the case. 
+
+Afterwards the node may try to *re-bootstrap* again (if the node did not recently *bootstrap*). Finally if the node still lacks peers then it will send out *GetPeers* messages to its peers.
+
+The node listens on its established connections and processes the received messages accordingly. If the message was *re-broadcastable*, then the node will relay the message to its own peers such that the message will propagate throughout the network. Beforehand processing the message the node asserts that it has not yet seen the message before or otherwise the node will not process the *duplicated* message. 
+
+The control flow of the *network layer* can be summarized as follows:
+
+1. Listen for incoming connections
+2. Process network messages
+3. Housekeeping
+4. (Optionally) Establish new connections
+5. Repeat
+
+## Banning
+A node holds information if another peer is *permanent banned* or *soft banned* (e.g. a temporary ban could have a duration of 24 hours). 
+Permanent banning is carried out by a manual process, thus the operator of the node can put other peers on the permanent ban list by calling the RPC API. Banned peers are identified by their IP address and so the node will not establish connections to such an address.
+
+*Soft banning* is on the other hand an automatic procedure carried out by the node. The node *soft bans* a peer based on the outcome from the *consensus* layer, reasons include; the peer sends malformed messages, the peer sends messages with *bad* data e.g. an inconsistent *catch-up* message etc. *Soft bans* are also given by the IP address of the peer and as such the node will immediately disconnect from the peer and avoid establishing new connections to the peer within the *soft ban duration*.
+
+## Connecting to the *Mainnet*
+In order to connect to the Concordium *Mainnet* the node should be configurged with the Concordium *bootstrappers*. Currently there exists 2 *bootstrappers* for the Concordium *Mainnet* and they can be reached at `bootstrap.mainnet.concordium.software`.
```

Comment:
If the mainnet ones are here, the testnet ones should also be `bootstrap.testnet.concordium.com`.